### PR TITLE
fix(pagination): default first and last if required

### PR DESCRIPTION
**Pull request by littlemooon**

### CODE REVIEW

File: packages/codegen-sdk/src/print-connection.ts [hunk 1]
```diff
@@ -51,7 +51,28 @@ function printAbstractConnection(): string {
...
+    }`,
+  ]);
+}
+
+/**
+ * Function to default any connection variables that are rerquired by the api
+ */
+function printConnectionDefault(): string {
+  return printLines([
+    printComment(["Default connection variables required for pagination", "Defaults to 50 as per the Linear API"]),
```
Comment (littlemooon) [inline packages/codegen-sdk/src/print-connection.ts:68]: Add a function to the sdk to default the first and last connection variables to 50 if not set

File: packages/codegen-sdk/src/print-operation.ts [hunk 2]
```diff
@@ -71,11 +71,13 @@ function printOperationCall(context: SdkPluginContext, operation: SdkOperation):
         `this._${Sdk.REQUEST_NAME}`,
         `${Sdk.CONNECTION_NAME} => this.${Sdk.FETCH_NAME}(${printList([
           ...operation.requiredArgs.args.filter(arg => !parentArgNames.includes(arg.name)).map(arg => arg.name),
-          `{ ${printList([
-            ...optionalArgs.map(arg => `...this._${arg.name}`),
-            `...${Sdk.VARIABLE_NAME}`,
-            `...${Sdk.CONNECTION_NAME}`,
-          ])} }`,
+          `${Sdk.CONNECTION_DEFAULT}({
```
Comment (littlemooon) [inline packages/codegen-sdk/src/print-operation.ts:74]: Wrap connection variables in the defaulting function

File: packages/sdk/src/_generated_sdk.ts [hunk 3]
```diff
@@ -27,7 +27,24 @@ export type LinearFetch<Response> = Promise<Response>;
...
+  before?: string | null;
+  first?: number | null;
+  last?: number | null;
+};
+
+/**
+ * Default connection variables required for pagination
+ * Defaults to 50 as per the Linear API
+ */
+function defaultConnection<Variables extends LinearConnectionVariables>(variables: Variables): Variables {
```
Comment (littlemooon) [inline packages/sdk/src/_generated_sdk.ts:41]: Default function output

File: packages/sdk/src/_generated_sdk.ts [hunk 4]
```diff
@@ -5660,7 +5687,14 @@ export class AttachmentsForUrlQuery extends Request {
     const data = response.attachmentsForURL;
     return new AttachmentConnection(
       this._request,
-      connection => this.fetch(url, { ...variables, ...connection }),
+      connection =>
+        this.fetch(
+          url,
+          defaultConnection({
```
Comment (littlemooon) [inline packages/sdk/src/_generated_sdk.ts:5693]: Wrap output

**Review by tommoor — APPROVED** [review]
Its beautiful

**Pull request merged**